### PR TITLE
bug: Increment transition_id if block changeset is present

### LINKED ISSUE #1655: TransitionId increment in execution vs blocks stage
### Describe the bug

Body and execution stage needs to have same rules on how to increment transition id for block changeset

Currently, execution stage is doing +1 always:
https://github.com/paradigmxyz/reth/blob/32e31b240dbfe9c98f6f7bfba9f76b878a3dc728/crates/storage/provider/src/transaction.rs#L733

While body stage has different logic:
https://github.com/paradigmxyz/reth/blob/32e31b240dbfe9c98f6f7bfba9f76b878a3dc728/crates/stages/src/stages/bodies.rs#L180

Block changeset was initially `Option` it probably changed on refactor 

### Steps to reproduce

Not happening atm

### Node logs

_No response_

### Platform(s)

_No response_

### What version/commit are you on?

_No response_

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by rakita**
closes #1655

### CODE REVIEW

File: crates/storage/provider/src/transaction.rs [hunk 1]
```diff
@@ -719,6 +719,13 @@ where
                 current_transition_id += 1;
             }
 
+            // Transition is incremeneted every time before Paris hardfork and after
+            // Shanghai only if there are Withdrawals in the block. So it is correct to
+            // to increment transition id every time there is a block changeset present.
+            if !results.block_changesets.is_empty() {
+                current_transition_id += 1;
+            }
+
```
Comment (rkrasiuk) [inline crates/storage/provider/src/transaction.rs:728]: i'm still thinking about possible repercussions of this. meanwhile, something seems wrong here. current transition id is not selected correctly

**Review by rkrasiuk — CHANGES_REQUESTED** [review]

File: crates/storage/provider/src/transaction.rs (hunk 1 above)
Comment (rakita) [reply crates/storage/provider/src/transaction.rs:728]: Wdym about id is not selected correctly?

Check initial issue, we were differently setting transition id, and this worked as every before Paris block had a block reward as changeset.

Between Paris and Shanghai, there isn't any block changesets and transition_id will get wrongly incremented

Comment (rakita) [reply crates/storage/provider/src/transaction.rs]: moved order, increment needs to be after insertion.

**Comment by rkrasiuk** [comment]
let's not merge this PR until we fix a bigger issue of inconsistencies with block transitions

**Comment by rakita** [comment]
I tried to fix the Body state tests, but couldn't manage to do in easy way.

**Review by rkrasiuk — APPROVED** [review]

**Comment by rakita** [comment]
Left comment here: https://github.com/paradigmxyz/reth/pull/1722#issuecomment-1465333286
about gaps that we talked in discord

**Comment by gakonst** [comment]
cc @onbjerg for visibility

**Pull request merged**